### PR TITLE
Add a file extension to download url

### PR DESCRIPTION
**Pull request by mjh1**
This relies on a CDN rule to re-route the request to the standard `/video` path. I'll probably put it behind an experiment to begin with and also probably should have the mapping of format to file extension in catalyst.

**What does this pull request do? Explain your changes. (required)**

**Specific updates (required)**

**How did you test each of these updates (required)**

**Does this pull request close any open issues?**

**Screenshots (optional)**

**Checklist**

- [ ] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/api/src/controllers/asset.ts [hunk 1]
```diff
@@ -418,7 +418,23 @@ function getDownloadUrl(
...
+      source.path != "video" ||
+      !formatToFileExtension.has(asset.videoSpec.format)
+    ) {
+      return pathJoin(base, asset.playbackId, source.path);
+    }
+    return pathJoin(
+      base,
+      asset.playbackId,
+      source.path,
+      "download." + formatToFileExtension.get(asset.videoSpec.format),
```
Comment (victorges) [inline packages/api/src/controllers/asset.ts:439]: wdyt of naming the file with the asset name instead?
```suggestion
      encodeURIComponent(`${asset.name}.${formatToFileExtension.get(asset.videoSpec.format)}`),
```

**Review by victorges — APPROVED** [review]

File: packages/api/src/controllers/asset.ts (hunk 1 above)
Comment (mjh1) [reply packages/api/src/controllers/asset.ts:439]: Only issue with that is that the name for direct uploads will contain the original file extension and so they'll end up with double file extensions :( I like the idea though, it is annoying when you end up downloading files all with the same name.
Screenshot 2024-09-19 at 12 21 13

Comment (mjh1) [reply packages/api/src/controllers/asset.ts:439]: could just strip anything after the final `.`? Or just use everything before the first `.`

**Comment by mjh1** [comment]
@thomshutt @victorges I'm also tempted to put this behind an experiment for now... although the old URLs will still work I feel like we probably need to announce somewhere we're making this change to the API

**Comment by thomshutt** [comment]
@mjh1 Let's not do an experiment, but you can work with @Shih-Yu on messaging and giving enough notice

**Pull request merged**